### ISSUE TITLE
Enhancement: Reorganize the FAQ

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/temp-test-ioccc#2682]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like it

### Describe the problem with or missing documentation

Now that links to FAQ items now involve URLs with “_#string_”s that do **NOT** involve FAQ numbers (apart from the FAQ table of contents),

and now that references to FAQ entries are of the form:

````
See the
FAQ on "[GitHub pull request](#pull_request)"
for more information about pull requests.
````

It is time to reorganize the FAQ into better sections and to rebuild FAQ table of contents accordingly.

### Relevant files

[faq.md](https://github.com/ioccc-src/temp-test-ioccc/blob/master/faq.md)

### What you expect

The **most important** FAQs will at the top of the FAQ.  These **most important** FAQs will be a select few relating to entering the IOCCC.

Other FAQs will be organized into suitable sections below.

### Relevant links or files

[faq.html](https://ioccc-src.github.io/temp-test-ioccc/faq.html)

### Anything else?

What is currently in FAQ “_Section 0_”, while all FAQ items in section 0 are useful, is **too long**.  Someone who is impatient to enter IOCCC28 should not have to wade through items such as “How to validate any JSON document”.  While that FAQ is informative, is unlikely to be needed by most people who are simply wanting to understand how to enter the IOCCC.

The current FAQ structure, while informative and potentially interesting to someone who is into the details of the IOCCC, is a potentially intimidating barrier to new people wishing to simply enter the IOCCC.

There should be a clear distinction between a few select and short “TL;DR” type FAQs relating to entering the FAQ, and those FAQs that relate to small or somewhat internal IOCCC details.

**Comment by lcn2** [comment]
# QUESTION

What are the **bare minimum** FAQs that are really needed by someone who wishes to enter the IOCCC?

The top FAQ section should just contain that **bare minimum** set.  Perhaps that top FAQ section be called something along the lines of "_TL;DR Entering the IOCCC - a quickstart guide_"?   Then a later FAQ section could contain other sections current section 0 stuff and be called something along the lines of "_More details about entering the IOCCC_".  Those are just suggestions.

The contents of the top **bare minimum** "_TL;DR_" set should be very short, preferably that fits within a "single screen".  If the contents of one of these "_TL;DR_" entries is too long, then more verbose part of that **bare minimum** FAQ should be moved into another FAQ (perhaps in the so-called "_More details about entering the IOCCC_" section), along with a "see also" link/reference to that more detailed content.

There is nothing wrong with an FAQ entry that goes into great detail with the FAQ.  Such long and detailed FAQs, however, should not be part of the **bare minimum** FAQ "_TL;DR_" set.

Obviously the rest of the FAQ set needs to be reorganized and perhaps retitled too.

Other long and detailed FAQs elsewhere in the FAQ may wish to be also broken up into two FAQs: one for the high level summary that contains a "see also" link/reference to that more detailed content in a subsequent FAQ.

**Comment by lcn2** [comment]
This issue is a **top priority** and is required to be completed before the **Great Fork Merge**.  This issue is essentially part of issue #2006 and if of equal importance to that work.

We broke this out from the critical **top priority** issue #2006 to make it easier to discuss FAQ related issues.

**Comment by xexyl** [comment]
Okay the reason I have not done this yet is because I figure I might end up having to modify the FAQ when I work on the entries that I have not yet finished.

The same goes for the bugs file though I'm also looking at that with each entry.

You can assign this to me though I agree it is part of #2006 and will be done after the entries' html files have been fixed/checked.

Now that we'll have a code freeze in mkiocccentry I hope to get back to this repo.

**Comment by xexyl** [comment]
Please add to the todo something about adding an entry to the FAQ about Zlib. The entry that requires it and should be updated to link to this new FAQ is 2012/vik.

I am aiming to finish 2012 today too. Not sure if I should go beyond that as I'm having a hard time seeing but my idea that I mentioned in the other one, about a deadline, is helping. 

Thanks!

**Comment by xexyl** [comment]
Also would you please add an FAQ about IOCCC BOF? I know what it stands for but that's about it - but you also refer to it in several README.md files so it'd be nice if it could be linked to.

Thanks.

**Comment by lcn2** [comment]
Regarding [GH-issuecomment-2409531657](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2409531657) and [GH-issuecomment-2409549155](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2409549155) see commit 4386b63a8d988b27515f43210e62d119863a1fad.

**Comment by xexyl** [comment]
> It is **VERY IMPORTANT** to understand and this issue is about reorganizing the table of contents and the sections under which existing FAQs reside.
> 
> This issue is **NOT** about rewriting entries, nor adding new FAQ entries. This issue is about reorganizing the existing set of FAQ entries. That includes the "TL;DR" section as suggested in the top level TODO.
> 
> Moreover, the default action should this issue be still open a day or two before 2024 Oct 29 is to leave the FAQ as it is now.

I understand this. But it felt better to have this done before I start doing this. I hope I can get to this issue but obviously the entries are of higher priority!

**Comment by xexyl** [comment]
Making good progress on reorganising the FAQ.

I am starting out by removing the anchors of the form `"faq[0-9]` (not regexp of course) so that it's easier to move them. After things are in order if it is desired to have that for each FAQ it can be added back. Doing it this way will prevent a mess and make it much easier.

It is not clear to me at this point that we even necessarily need to number them since the text we link to is always a name. However the number could be of use anyway. Certainly the sections should be numbered which might suggest that each one should be numbered. The downside to that is it makes it harder to add a new FAQ that might fit above another in its section.

I'll be going afk now but when back I'll continue. Once I have the links sorted (involving making new anchors too) I can test them on my server and then if that's good then I can worry about reordering things (possibly with new sections - or not .. that has to be determined still).

I do hope to finish this or mostly finish this today. I should have extra time today but I unfortunately also woke up earlier, albeit only half an hour earlier. Was up a bit later (or maybe it just felt like it, especially as I am more tired than usual) last night and likely tonight too but I probably was actually asleep at the usual time, it just feels later.

**Comment by xexyl** [comment]
> 
> ### Anything else?
> What is currently in FAQ “_Section 0_”, while all FAQ items in section 0 are useful, is **too long**. Someone who is impatient to enter IOCCC28 should not have to wade through items such as “How to validate any JSON document”. While that FAQ is informative, is unlikely to be needed by most people who are simply wanting to understand how to enter the IOCCC.

I agree that many are quite long. Some need to be - if we want to keep them. Do we need to document the `.info.json` and `.auth.json` files? What about the `.entry.json` files? Maybe yes, maybe no. Those FAQs are quite long though and might benefit with a `TL;DR`, if we keep them.

# QUESTION

I think it might be g good to remove the how to validate the JSON FAQ. If you agree I'll do that but not now.

Do you think it should be removed? It kind of seems out of scope to me.

I'm working on the anchor names. If that goes well I will then work on reordering things. Whether I do a pull request when it's incomplete (reordering wise) I do not know.

As I noted I am for now removing the numbers as we don't refer to them anyway and it is a burden when trying to reorganise and worse - mostly important later on - adding to the list, if that ever is necessary. I guess it might be necessary though most likely it'll be rare.

> 
> The current FAQ structure, while informative and potentially interesting to someone who is into the details of the IOCCC, is a potentially intimidating barrier to new people wishing to simply enter the IOCCC.
> 

That's true. The question is what should be removed and what should be kept? The ordering of course is another matter entirely and will take quite some work.

> There should be a clear distinction between a few select and short “TL;DR” type FAQs relating to entering the FAQ, and those FAQs that relate to small or somewhat internal IOCCC details.

Please elaborate on this. A clear distinction between ... I'm not clear on what you're getting at. 

Thanks.

<hr>

I do want to make sure I can work on the bugs file and clean up (and hopefully shorten, perhaps a lot) the thanks file, too, but I guess the FAQ is far more important so I'm trying to work on it first. Nonetheless it's going to take a tremendous amount of work, unfortunately, and I'm running out of time to meet the deadline.

**Comment by xexyl** [comment]
I just saw a picture of .. F. D. C. Willard. A Chocolate Point Siamese (or it looks like a Siamese)!

I also have a Chocolate Point Siamese and a Lynx Point as well (her son). The Chocolate Point is called Brownie but not strictly because she's a Chocolate Point. The reason is the first week we got her she was hiding (she was found outside a pound with her babies, starving herself to make sure they were okay, so she's always been extra cautious/timid). So not only the Chocolate Point but also brownies in folklore.

Here's a picture of her:

![IMG_6258](https://github.com/user-attachments/assets/26a168b0-74cc-46f5-90da-a0cdcd7e6009)

She hasn't been on the stair rail in a long time; this picture was taken 2020. Here's Smokey:

![IMG_7747](https://github.com/user-attachments/assets/57dc5a5c-89df-47f7-ac78-481898876c28)

That was taken 10 October 2024.

Fun words for you:

- ailurophile - lover of cats
- ailurophobe - person afraid of cats
- ailurophobia - fear of cats
- ounce - another term for snow leopard (no not making that up)
- destruction - group of wild cats (no not making that up .. neither am I when I say a group of rhinos is called a crash)

**Comment by lcn2** [comment]
> I think it might be g good to remove the how to validate the JSON FAQ. If you agree I'll do that but not now.
> 
> Do you think it should be removed? It kind of seems out of scope to me.
> 
> 

We agree that it is out of scope.  Please remove and please remove any links/references to that FAQ.

**Comment by lcn2** [comment]
> There should be a clear distinction between a few select and short “TL;DR” type FAQs relating to entering the FAQ, and those FAQs that relate to small or somewhat internal IOCCC details.
> Please elaborate on this. A clear distinction between ... I'm not clear on what you're getting at.
> 
> Thanks.

Someone who just wants to know the basics of participating / submitting to the IOCCC does **NOT** want to wade through a lot of FAQ detail, however interesting and informative those FAQ details might be to someone.

The "TL;DR" section (or whatever it is called) should contain just a few critical FAQ entries with only a brief sentence or 2 or 3.  Short and to the point FAQ entries with just the high level details.  These critical FAQs can, if needed, have a "see also link" to a lower down FAQ that can provide mode tail if needed.

Critical / high level / "TL;DR" FAQs should be at the very top of the FAQ, just a few of then, and be short and easy to read, light on details and to the point.  Someone looking at the overall FAQ for the first time should have their eyes drawn to these new top "TL;DR" FAQs AND the remaining FAQs should be farther down in the FAQ file.  There should be a clear header separation between the few critical  "TL;DR" at the top, and the rest of the FAQ.  Such a header separation could say something like "FAQs for those curious about the finer details", or something like that.

What should be in the critical "TL;DR"?  Here is an initial guess:

- how to I submit my code the IOCCC?

- how do I register for the IOCCC?

- what are the IOCCC rules and guidelines?

- ???

**Comment by xexyl** [comment]
BTW: ![image](https://github.com/user-attachments/assets/0db8845a-267d-4c70-b59e-f460f9bd1bd2)

**Comment by xexyl** [comment]
I will reply to your comment later but possibly not until tomorrow. But I will have more time today. I just have to do something else first.

**Comment by xexyl** [comment]
> > I think it might be g good to remove the how to validate the JSON FAQ. If you agree I'll do that but not now.
> > Do you think it should be removed? It kind of seems out of scope to me.
> 
> We agree that it is out of scope. Please remove and please remove any links/references to that FAQ.
> 
> ## UPDATE 0
> > Perhaps it belongs as a footnote to one of the other FAQs? I don't know though: might not be possible to make it short enough as a foot note.
> 
> That is a good option too.

Hmm. Or perhaps the jparse repo should have an FAQ and it can link to that instead? Or if not an FAQ a section in the README.md? That would keep it clean and simple and those who really wanted to know could look at it. But the vast majority of people will only care about validating IOCCC specific JSON files and so when the chkentry tool is updated later on nobody will even need to use the jparse tool (itself).

What do you think about a link instead? I could just as well delete it though. I don't really mind: it might be useful to some who want to know but the FAQ is too long even if it has a lot of good content. Even if some people have no problem reading a lot (like us for example) that does not mean they would want to; they would rather read something that they enjoy. They want to find their question quickly and get the answer they need.

So either a link or delete that entry entirely. The only reason to keep the jparse reference there is due to `make test`. But most don't need to know that so this is why I suggest at most a link. Nothing more but perhaps less.

**Comment by xexyl** [comment]
> > There should be a clear distinction between a few select and short “TL;DR” type FAQs relating to entering the FAQ, and those FAQs that relate to small or somewhat internal IOCCC details.
> > Please elaborate on this. A clear distinction between ... I'm not clear on what you're getting at.
> > Thanks.
> 
> Someone who just wants to know the basics of participating / submitting to the IOCCC does **NOT** want to wade through a lot of FAQ detail, however interesting and informative those FAQ details might be to others.
> 
> The "TL;DR" section (or whatever it is called) should contain just a few critical FAQ entries with only a brief sentence or 2 or 3. Short and to the point FAQ entries with just the high level details. These critical FAQs can, if needed, have a "see also link" to a lower down FAQ that can provide mode details if needed.
> 
> Critical / high level / "TL;DR" FAQs should be at the very top of the FAQ. There should be just a few of them. The need to be short and easy to read, light on details and to the point.
> 
> Someone looking at the overall FAQ for the first time should have their eyes drawn to these new top "TL;DR" FAQs AND the remaining FAQs should be farther down in the FAQ file. There should be a clear header separation between the few critical "TL;DR" at the top, and the rest of the FAQ. Such a header separation could say something like "FAQs for those curious about the finer details", or something like that.
> 
> The "TL;DR" section name is also open for discussion: that is just a working title that may or may not be good enough for the final FAQ file.
> 
> The critical "TL;DR" section should be very short. Perhaps short enough to fit on a standard desktop screen (of reasonable size), if that is at all possible?
> 
> What should be in the critical "TL;DR"? Just a select few. Here is an initial guess (these are not in any particular order):
> 
> * What is the IOCCC?
> * how to I submit my code the IOCCC?
> * how do I register for the IOCCC?
> * what are the IOCCC rules and guidelines?
> * Where to go to get more help?
> 
> ## UPDATE 0
> # QUESTION
> What do you think are the critical "TL;DR" FAQs? That is, what are the few things that someone might need to know about the IOCCC?


Before I give my feedback or even answer your question I want to make sure I follow you right. Then tomorrow morning I can reply (though I do have a phone call in the morning - doctor check-up, but likely I can reply before this as I wake up before it's even light out).

So... are you suggesting that there is a new section in the FAQ that are the most important things, like the list you give above, and then below that can be more detailed information (which some of the tl;dr FAQs might actually link to - or not)?

If you don't mean that what do you mean please? (I think the list you gave is reasonable to start out with though certainly but depending on if I follow you right, and I'm pretty sure I do but want to be certain, I might not have many suggestions.)

Thanks and best wishes on your recovery! I'll be back tomorrow.

**Comment by lcn2** [comment]
> So... are you suggesting that there is a new section in the FAQ that are the most important things, like the list you give above, and then below that can be more detailed information (which some of the tl;dr FAQs might actually link to - or not)?

Yes.  See [GH-issuecomment-2418145374](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2418145374) for details.

**Comment by xexyl** [comment]
> > So... are you suggesting that there is a new section in the FAQ that are the most important things, like the list you give above, and then below that can be more detailed information (which some of the tl;dr FAQs might actually link to - or not)?
> 
> 
> 
> Yes.  See [GH-issuecomment-2418145374](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2418145374) for details.

This link is to my comment about validating JSON documents?

I guess you meant to make it the one I was replying to initially?

I will look at it sometime this morning. In a short bit of time I have to go away for a bit but I hope to make much more progress today on the FAQ. 

I also want to quickly look at an entry of mine to see if I can simplify the try section. 

Anyway now that you confirmed my thought I will be able to address your questions better. 

Hope you're feeling better today!

**Comment by xexyl** [comment]
I'm currently pondering what to call the top level section ... going afk a bit. I'll have some thoughts later on. One candidate is 'How to enter the IOCCC: the bare minimum'.

But it might be also useful or a new document, somewhere, if possible? I'm not sure where it would fit though, so maybe not. I do think that the guidelines and rules should both link to the FAQ, however, probably towards the top. I can do that too.

These are general notes for me. I doubt I'll be afk long if at all but even if I'm not afk now I will be afk in a little bit.

**Comment by lcn2** [comment]
> But it might be also useful or a new document, somewhere, if possible? 

No new files, please. Now is not the time to start adding new FAQ files. 

> I'm not sure where it would fit though, so maybe not. I do think that the guidelines and rules should both link to the FAQ, however, probably towards the top. I can do that too.

Please 🙏 do not overthink this.    Keep it simple please.

The "TL;DR" is simply the top block of FAQs with a name that implies these are the quick or most common FAQs for those who wish to join the IOCCC .. followed by a visual separator (such as a level one heading), followed by the rest of the FAQ. 

Something like that.  It can be done in markdown.  Nothing special needed.

**Comment by xexyl** [comment]
No worries. Already figured it out. Will have a pull request later on.

**Comment by xexyl** [comment]
Once you merge https://github.com/ioccc-src/temp-test-ioccc/pull/2702/commits/330ebccc76c3e5a0f1f92573110d5eb149a1357e and can look at the table of contents we can go from there.

I did notice that some sections could probably be removed or greatly simplified. For instance the how to enter section talks about the mkiocccentry toolkit so couldn't the sections of what it is and how to use it be reduced somewhat? I don't know. On the one hand I'm afraid to not have enough information but on the other too much can be a problem. But then the guidelines also talk about how to use it so maybe this is not needed? Well we can look at it and discuss it.

I have to do something else now. If I have time I will look at the bugs or thanks file. Otherwise I'll continue tomorrow. I guess that's what'll happen.

**Comment by xexyl** [comment]
Please take a look at the new ordering of questions and let me know if you think anything should change. 

Another thing crossed my mind but unfortunately it is not coming to me right now. 

But if you get back on this tonight I can look at it tomorrow morning. 

Of course if you want and can and iff you have the time and it's not too much to do you MAY move the questions to the right sections but I can also do that. 

QUESTION 

How should each question be labelled? For instance should it be prefixed with 'Q: '? Having numbers makes it harder to update but I guess it could be done after rearrangement is finished. 

I am not looking at it now so I can't comment specifically but I do think that some questions could be either removed outright or drastically simplified. But determining which ones is complicated. 

I am trying to get this done but I also have two other files that I need to look at. 

Anyway hopefully after looking at the new sections and table of contents you will have some additional thoughts. 

Thanks for feedback!

Have a great night!

**Comment by lcn2** [comment]
> Please take a look at the new ordering of questions and let me know if you think anything should change.

At first glance `faq.html` looks good.  Thank you @xexyl!

Removing all of the FAQ X.Y from the table of contents makes it much easier to reorder them.

> Another thing crossed my mind but unfortunately it is not coming to me right now.
> 
> But if you get back on this tonight I can look at it tomorrow morning.
> 
> Of course if you want and can and iff you have the time and it's not too much to do you MAY move the questions to the right sections but I can also do that.
> 
> QUESTION
> 
> How should each question be labelled? For instance should it be prefixed with 'Q: '? Having numbers makes it harder to update but I guess it could be done after rearrangement is finished.

The Section markdown headers such as:

````
## Section 0: Entering the IOCCC: the bare minimum you need to know
```

is a good idea.

For the action FAQ entries, perhaps your idea of:

````
Q: How can I enter the IOCCC?
````

is a good idea.  Alternately the "Q:" could become "QUESTION:". 

And certainly not "FAQ 0.16 -" as most are now.

Your choice.

A tricky thing to keep is that the table of contents order and the FAQs under the given section in the same order.  We will need to be careful about this in the future.

**Comment by xexyl** [comment]
> > Please take a look at the new ordering of questions and let me know if you think anything should change.
> 
> At first glance `faq.html` looks much improved. Thank you @xexyl, this is very good progress.
> 

Yes although (as noted below) I did **NOT** move the actual questions to the right order.

> > Another thing crossed my mind but unfortunately it is not coming to me right now.
> > But if you get back on this tonight I can look at it tomorrow morning.
> > Of course if you want and can and iff you have the time and it's not too much to do you MAY move the questions to the right sections but I can also do that.
> > QUESTION
> > How should each question be labelled? For instance should it be prefixed with 'Q: '? Having numbers makes it harder to update but I guess it could be done after rearrangement is finished.
> 
> The Section markdown headers such as:
> 
> ```
> ## Section 0: Entering the IOCCC: the bare minimum you need to know
> ```
> 
> is a good idea.
> 

I think it works for the name too.

> For the action FAQ entries, perhaps your idea of:
> 
> ```
> <div id="mkiocccentry">
> ### Q: What is the `mkiocccentry` tool and how do I use it?
> </div>
> ```
> 
> is a good idea. However ...

On the other hand so is `### FAQ: ...`. I guess. I don't know.

> 
> ## UPDATE 0
> A tricky thing to keep is that the table of contents order and the FAQs under the given section in the same order. We will need to be careful about this in the future.

Yes.
> 
> Please check this. For example the "What types of entries have been frequently submitted to the IOCCC?" is in the wrong order.

Yes. As noted earlier that's because I did not want to worry about it until after the order of the table of contents is correct. So this was very intentional to save time. No sense doing it more than once.

> 
> Sadly this is a consequence of not using "FAQ x.y -" in the entries.
> 

Actually it's not that. It's just that I did not bother to do that - yet. The problem with having numbers is it's harder to update. Makes it so if the order has to change all the numbers after it have to be changed too which is tedious and error prone. If one does not do it often it might be okay but still a burden.

Regardless the numbers should not be there until after the list is done.

# QUESTION

It is a shame we cannot more easily use `<ol>`. Well we could in the table of contents now I think on it. But to do it in the FAQ itself we would have to add a `<li>` and `<ol>` and `</ol>` and `</li>` in the document itself. Maybe that would be worth it. What do you think?

> Another problem we see is that when one jumps down into the middle of the FAQ, on can get lost and not know which section the FAQ is in.
> 

That's true.

> Perhaps instead of using "Q:" we should use "Q x.y:"? We know this is a reversal of a previous idea, however:
> 
> * it is easy to get lost in the middle of the FAQ
> * it is hard to keep the FAQs in the same order as the table of contents
> 

Well problem one is going to happen regardless.

Problem two is hard to make it the same order whether or not it's numbered. It's easy to find out which it should be under/above but it's tedious to actually move it. But one method of finding it is open it in the browser and then edit in in vim or in this case, since it takes some time (unless you have it in a reasonable enough order so you can do visual mode and then search to go down to it), use something like TextEdit. I actually did this for some parts yesterday until I had the thought that I should wait until the table of contents is sorted.



> So perhaps we need:
> 
> ```
> <div id="mkiocccentry">
> ### Q 0.1: What is the `mkiocccentry` tool and how do I use it?
> </div>
> ```

Maybe. See above for a thought that might make it easier. It would require more work to set up but once it's set up it would be ordered. Though I think the `<ol>` does not let one use 0 for the first number.

I can play with this in the table of contents and the sections too. A short section would be a good test case.

> 
> ### QUESTION about UPDATE 0
> What do you think about this problem and how to solve it?
> 

Depends on what problem. I gave a possible solution to the numbering problem. It would take a bit of time to set up but it would solve the annoying problem of ordering. Kind of. It wouldn't have the `Q: x.y` but it would have the numbered items.

I think that might be worth it. It would also make it easier in one way: we would still have to put the questions in the right order but we would not have to update every number manually.


> Should the entries in the table of contents also start with "Q x.y:"?
> 

I don't know. Using ordered lists like I suggested would simplify this. I'll play with this today.

> NOTE: If we do put "Q x.y" we would still NOT link to some "#x_y" tag. We would link to names such as "#mkiocccentry". Moreover when we refer/link to some FAQ, we would NOT use a "Q x.y". For example:
> 

Oh definitely true. That's why I changed it the other day.


> ```
> See the
> FAQ on "[What is the `mkiocccentry` tool and how do I use it](#mkiocccentry)".
> ```
> 
> I.e., no "Q x.y" in the link name.
> 

Yes.

> Given the problem of keeping the Table of Contents in the same order as the FAQ, and given the problem of getting lost in the middle of the FAQ, we think we need the "Q: x.y" in both the table of contents and the FAQ markdown level 3 text.
> 

Sure but maintaining it is a pain. Unless my idea of ordered lists works well.


> We know this is a reversal of a previous idea, but unless you have a better idea of how to solve this problem, we might have to do this.
> 

See above.

> Again, putting "Q x.y:" in the table of contents and in the FAQ markdown level 3 text does **NOT** mean we would use "faq_x_y" link tags, nor would we refer to an FAQ elsewhere by the "x.y".
> 

Agreed.

> ## UPDATE 1
> Some of the entries in the table of contents do not end in a question mark ("?").
> 

Good catch. I'll fix that today.

> ## UPDATE 2
> Section 3 "Compiling and running IOCCC entries" is too long.
> 

Hmm ... I actually agree with this.

> We recommend this section should be divided into 2 sections somehow.
> 

Well using html lists would solve this problem, I think, as you can have lists in lists.

How would they be named though? My guess is that the sections would be (content wise, not name .. have to think about names):

0. Compiling the entries
1. Prerequisites to compile some entries.

Then another section would be:

Running the entries. Whether this is a subsection or not, or even whether the above two are subsections (perhaps not?) is unclear.

Maybe the names could be:

- How to compile winning IOCCC entries
- Resolving any dependencies (or 'Dependencies for some entries'? That sounds less good but the first one sounds kind of bad too)
- Running entries

Something like that.

As for making them subsections. I'm unsure. I think that dependencies belongs in the compiling section but perhaps the running the entries should be in its own section. The only reason to make the other one a subsection (or if it looks better) a new section is indeed because of the length. I'm mixed on the first two: I can see it both ways and the more I think on it the more I think making it another section might be a better way: exactly because it's already so long.


> ## UPDATE 3
> At the very bottom there are two "Jump to: top" entires. We suggest the short line and the end "Jump to: top" at the very end should be removed.
> 

Very end of the DOCUMENT? Or of each section? As long as the end of each FAQ has a jump to top I don't see why the end of the document should have one too.

> ## UPDATE 4
> We think that Section 0 should have fewer entries.
> 
> Perhaps we need a "Section 1 - More entering the IOCCC details".
> 

Perhaps the wording would be `Section 1: more details on entering the IOCCC`[0]

[0] I think that the table of contents should have the same character separator as the actual sections themselves. Whether that is `:` or `-` is unclear. I think the `:` looks nicer but that's me.

> Then some of the Section 0 FAQs can be moved in that new Section 1.
> 
> For example, consider moving from Section 0 in the newly proposed "Section 1 - More entering the IOCCC details" these entries:
> 
> * Is there a way to not have to re-enter the same information, when making a change to my submission?
> * May I use a different source or compiled filename than prog.c or prog?
> * What are the IOCCC best practices for using markdown?
> * How do I report bugs in an mkiocccentry tool?
> 
> Those FAQ are needed, but we need to keep the Section 0 set small.

I like that too.

Great feedback - thanks!

I'll play with the ordered lists soon (I hope). If this works then it's just a matter of having to set it up and then putting the questions in the correct order. But once that's done it'll be far easier to maintain.

**Comment by lcn2** [comment]
>> UPDATE 3
>>
>> At the very bottom there are two "Jump to: top" entires. We suggest the short line and the end "Jump to: top" at the very end should be removed.
>>
> Very end of the DOCUMENT? Or of each section? As long as the end of each FAQ has a jump to top I don't see why the end of the document should have one too.

"At the very bottom there are two "Jump to: top" entires. "

<img width="774" alt="Screenshot 2024-10-18 at 06 53 59" src="https://github.com/user-attachments/assets/6e3110b9-a22a-4937-8256-a97d3ffc5e07">

**Comment by xexyl** [comment]
> > > UPDATE 3
> > > At the very bottom there are two "Jump to: top" entires. We suggest the short line and the end "Jump to: top" at the very end should be removed.
> > 
> > 
> > Very end of the DOCUMENT? Or of each section? As long as the end of each FAQ has a jump to top I don't see why the end of the document should have one too.
> 
> "At the very bottom there are two "Jump to: top" entires. "
> 
> <img alt="Screenshot 2024-10-18 at 06 53 59" width="774" src="https://private-user-images.githubusercontent.com/10429707/377893619-6e3110b9-a22a-4937-8256-a97d3ffc5e07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNjAwNDYsIm5iZiI6MTcyOTI1OTc0NiwicGF0aCI6Ii8xMDQyOTcwNy8zNzc4OTM2MTktNmUzMTEwYjktYTIyYS00OTM3LTgyNTYtYTk3ZDNmZmM1ZTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDEzNTU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxMThjNGNlOTY0YTJlYmFkMjg3ZTA5OTI3ZTkxMjhlZjkxMmY2ODUyNmU4YTk4MDJhNDI1NWUyYjk2MzdjODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-Z0kca7_JrZ2cBR_yeo2uoVY0SWgNuEmdcyWaItutk4">

So I thought. Thanks! Will do.

**Comment by xexyl** [comment]
I like the way the ordered list works! Check this screenshot:

![Screenshot 2024-10-18 at 06 57 04](https://github.com/user-attachments/assets/9de05086-8cb8-4982-89e8-f6e505f783fa)

This will greatly simplify things. All we have do once this is completely set up is to make sure in the content (not table of contents: the actual answers) add the appropriate `<li></li>` and at the end of each section have `</ol>` and beginning of each section have the `<ol>` and make sure to put the answer in the correct location.

I'll set this up today.

**Comment by lcn2** [comment]
> Sure but maintaining it is a pain. Unless my idea of ordered lists works well.

Making and ordered list does not help as the Table of Contents order would be auto numbered in one order while the FAQ themselves might be auto numbered differently.

At least if you make in a "Q 2.4:" string in to both the table of contents and the FAQ, you set the order once correctly once and leave it that way.   You set the order in the Table of Contents with "Q X.Y:" first, then jump down to the FAQ and set the same string there.


  The ordered lists will renumber of things move around.

**Comment by lcn2** [comment]
> I like the way the ordered list works! Check this screenshot:
> 
> ![Screenshot 2024-10-18 at 06 57 04](https://private-user-images.githubusercontent.com/53008573/377894780-9de05086-8cb8-4982-89e8-f6e505f783fa.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNjAyMzksIm5iZiI6MTcyOTI1OTkzOSwicGF0aCI6Ii81MzAwODU3My8zNzc4OTQ3ODAtOWRlMDUwODYtOGNiOC00OTgyLTg5ZTgtZjZlNTA1Zjc4M2ZhLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDEzNTg1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiZjcxOWU1M2I4MzJiZmJiYjNmODNiODc5NjI0ZTQ3ZDc4NTNhNDllMjEzNjA3ZTlkNWZhNmFmNDA0MjgzNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HRB0aSiQAd2kBO7eWLflTG7WBUXnEwS4I2WbJeSzSn0)
> 
> This will greatly simplify things. All we have do once this is completely set up is to make sure in the content (not table of contents: the actual answers) add the appropriate `<li></li>` and at the end of each section have `</ol>` and beginning of each section have the `<ol>` and make sure to put the answer in the correct location.

You need a 2 level set of numbering, not just a single number.

**Comment by xexyl** [comment]
> > Sure but maintaining it is a pain. Unless my idea of ordered lists works well.
> 
> Making and ordered list does not help as the Table of Contents order would be auto numbered in one order while the FAQ themselves might be auto numbered differently.
> 
> At least if you make in a "Q 2.4:" string in to both the table of contents and the FAQ, you set the order once correctly once and leave it that way. You set the order in the Table of Contents with "Q X.Y:" first, then jump down to the FAQ and set the same string there.
> 
> The ordered lists will renumber of things move around.

It will. But without ordered lists we have to manually maintain it and also do make sure the order is correct. With ordered lists we just have to put the answer in the correct place.

**Comment by xexyl** [comment]
> > I like the way the ordered list works! Check this screenshot:
> > ![Screenshot 2024-10-18 at 06 57 04](https://private-user-images.githubusercontent.com/53008573/377894780-9de05086-8cb8-4982-89e8-f6e505f783fa.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNjAyMzksIm5iZiI6MTcyOTI1OTkzOSwicGF0aCI6Ii81MzAwODU3My8zNzc4OTQ3ODAtOWRlMDUwODYtOGNiOC00OTgyLTg5ZTgtZjZlNTA1Zjc4M2ZhLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDEzNTg1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiZjcxOWU1M2I4MzJiZmJiYjNmODNiODc5NjI0ZTQ3ZDc4NTNhNDllMjEzNjA3ZTlkNWZhNmFmNDA0MjgzNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HRB0aSiQAd2kBO7eWLflTG7WBUXnEwS4I2WbJeSzSn0)
> > This will greatly simplify things. All we have do once this is completely set up is to make sure in the content (not table of contents: the actual answers) add the appropriate `<li></li>` and at the end of each section have `</ol>` and beginning of each section have the `<ol>` and make sure to put the answer in the correct location.
> 
> You need a 2 level set of numbering, not just a single number.

Not sure why here but you can also do this with ordered lists. Done it many times.

**Comment by lcn2** [comment]
> Not sure why here but you can also do this with ordered lists. Done it many times.

Fine with a "x.y" 2-level number ordered list.

You will still need to order the FAQs in the same way regardless.

**Comment by xexyl** [comment]
> > Not sure why here but you can also do this with ordered lists. Done it many times.
> 
> Fine with a "x.y" 2-level number ordered list.
> 
> You will still need to order the FAQs in the same way regardless.

Yes. You have to put the answers in the same location but you don't have to maintain the numbers. Just put them in the correct location and if it's in between others in the section you do not have to change all the numbers after it.

**Comment by xexyl** [comment]
The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?

**Comment by lcn2** [comment]
> The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?

That's why your order list should be "X.Y", not just a counter.

**Comment by xexyl** [comment]
Now I can certainly roll that back but I don't fancy having to add the. numbers back. Certainly I won't do it until after the rest is done. But it seems like maintainability is far more important than having the `x.y` format.

**Comment by xexyl** [comment]
> > The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?
> 
> That's why your order list should be "X.Y", not just a counter.

Yes but afaik that's not possible with html :(

Or am I unaware of something ?

**Comment by lcn2** [comment]
> > > The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?
> > 
> > 
> > That's why your order list should be "X.Y", not just a counter.
> 
> Yes but afaik that's not possible with html :(
> 
> Or am I unaware of something ?

Does markdown have an option?

**Comment by xexyl** [comment]
> > > > The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?
> > > 
> > > 
> > > That's why your order list should be "X.Y", not just a counter.
> > 
> > 
> > Yes but afaik that's not possible with html :(
> > Or am I unaware of something ?
> 
> Does markdown have an option?

Not that I know of. Downside with it too is for markdown numbered lists you have to put the numbers in which you might as well do what we had before. So it would be no gain anyway, right?

**Comment by xexyl** [comment]
> > > > > > The only downside is that in the answers themselves you don't see the section number per question. I think that might be worth it?
> > > > > 
> > > > > 
> > > > > That's why your order list should be "X.Y", not just a counter.
> > > > 
> > > > 
> > > > Yes but afaik that's not possible with html :(
> > > > Or am I unaware of something ?
> > > 
> > > 
> > > Does markdown have an option?
> > 
> > 
> > Not that I know of. Downside with it too is for markdown numbered lists you have to put the numbers in which you might as well do what we had before. So it would be no gain anyway, right?
> 
> [perhaps this?](https://meta.stackexchange.com/a/85476)

But we would still have to put the numbers in manually so it would gain us nothing.  That's the problem.

**Comment by lcn2** [comment]
Well try the HTML if that doesn't work.

**Comment by xexyl** [comment]
> Well try the HTML if that doesn't work.

I'll do that. I have to go afk a bit. Maybe I can think of a hack. I also thought of another thing I have to do today. A medical thing, unfortunately. Fortunately it won't take that long though.

**Comment by xexyl** [comment]
I thought of a hack but unfortunately it doesn't work. The link you provided also in a comment suggests markdown doesn't support it but it would have the same problem as I noted anyway.

I thought of...

```markdown
0\.<li><a class="normal" href="#submit">How can I enter the IOCCC?</a></li>
```

The `\` was to try and escape any markdown rendering of it to make it a new line but unfortunately it doesn't work:

![Screenshot 2024-10-18 at 07 39 49](https://github.com/user-attachments/assets/790ddbba-396a-4f1a-82fd-e8bd979ba0ae)

.. which is awful and useless. Maybe some CSS would allow to fix it?

**Comment by xexyl** [comment]
Okay the menu is organised better with numbers. Next step is to restructure the sections a bit also in the menu. After that I can make sure the answers are in the correct order. Once this is done I can then add the list items part. This is very tedious but I hope it'll be good. I also wish there was a way to do the `x.y` format but I do not know how sadly. They really should add that as a feature but even if they did it would take a long while (I guess) for it to be in most browsers.

**Comment by xexyl** [comment]
Please let me know how the new menu and answer/question numbering looks when commit b62dc414789a10f4034660a79784ff7531169994 is merged. Unfortunately I have a number of other things on my mind that makes it hard to do the next step of moving the questions to the right section and even adding the in the questions the new sections. So this is a big work in progress and just deals with the menu.

I do like how it looks but links have to be fixed too and some FAQs can be either removed or reduced in length drastically. I think I might have to do the bugs file sometime soon, maybe next, to make sure it gets done as the FAQ work is taking a lot of time and is quite error prone (it turns out).

**Comment by xexyl** [comment]
I got the first section (0) so that the questions of the table of contents are in the order that the questions are in the actual section. There still are other questions below those in that section though but this is all I can do today with it.

I'm finding it extremely difficult to do the moving the stuff about so I have to break from this. Maybe it'll give you some time to look at it too and think about it. Maybe even I should wait so you can see if you want the order different. That was my intention in the first place but it slipped my mind.

So...

QUESTION

How does the menu look now? Should anything be changed? If all seems good I can work more on this tomorrow. Though I must admit I kind of dread the going back and forth throughout the file and copy pasting things. It's so error prone and I don't want to cause a big mess. It might be good if I were to work on the other files first or maybe look at trying to shorten some questions and then go back to this. But anyway let me know how the menu looks and if you think anything else should change. We can discuss this before I move more entries about.

Thanks and have a great rest of your day!

**Comment by lcn2** [comment]
QUESTION
> 
> How does the menu look now? Should anything be changed? If all seems good I can work more on this tomorrow. Though I must admit I kind of dread the going back and forth throughout the file and copy pasting things. It's so error prone and I don't want to cause a big mess. It might be good if I were to work on the other files first or maybe look at trying to shorten some questions and then go back to this. But anyway let me know how the menu looks and if you think anything else should change. We can discuss this before I move more entries about.

The concept in the Table of Contents looks reasonable.

Of course, and you probably implied it, we suggest you try some CSS work to further improve it.

You use of `class="normal"` might allow you to use a different class for some CSS might **LATER ON**.

We say **LATER ON** because "font-ing and shaping the appearance" should come after the file structure has been settled down.  So for now we will mostly ignore "font-ing and shaping the appearance" issues.

> Thanks and have a great rest of your day!

# SUGGESTION

When it comes to numbering the actual FAQ entries, you might want to continue your `class="normal"` trick so that you can later do "font-ing and shaping the appearance" as well.

We recommend you proceed.

**Comment by xexyl** [comment]
> QUESTION
> 
> > How does the menu look now? Should anything be changed? If all seems good I can work more on this tomorrow. Though I must admit I kind of dread the going back and forth throughout the file and copy pasting things. It's so error prone and I don't want to cause a big mess. It might be good if I were to work on the other files first or maybe look at trying to shorten some questions and then go back to this. But anyway let me know how the menu looks and if you think anything else should change. We can discuss this before I move more entries about.
> 
> The concept in the Table of Contents looks reasonable.
> 
> Of course, and you probably implied it, we suggest you try some CSS work to further improve it.
> 
> You use of `class="normal"` might allow you to use a different class for some CSS might **LATER ON**.
> 
> We say **LATER ON** because "font-ing and shaping the appearance" should come after the file structure has been settled down. So for now we will mostly ignore "font-ing and shaping the appearance" issues.
> 
> > Thanks and have a great rest of your day!
> 
> # SUGGESTION
> When it comes to numbering the actual FAQ entries, you might want to continue your `class="normal"` trick so that you can later do "font-ing and shaping the appearance" as well.
> 
> We recommend you proceed as you are doing with the rest of the document, such as to "move around the FAQ entires to match the Table of Contents" order, HTML list numbering, etc.

Thanks. That's what I needed to know: though the class=normal was your idea some time back.

I am afraid I'm not going to meet the deadline at this rate though :( so few days, so few days ...

I might want to look at the bugs file today or the thanks file or maybe both to get them out of the way, perhaps with working on the bugs file some too.

CSS is certainly something that might be of use though I don't know how (yet?).

I got some great night sky pictures last night[0]. I want to sort through them today too but I should be at the computer more time than usual today so if I can have enough energy I will have more time to work on things. Problem is last four nights have been rough: waking up more often and then the last two nights waking up earlier (than 5). 

[0] Yes I was hoping to see the comet. First night I tried (the other night) it was way too foggy. The next night it was foggy too. Then last night it was partly cloudy :( Still got some gorgeous pictures, both of Sunset (well technically it was after Sunset but the Sun was still visible .. it's not a science I understand that well actually, or maybe the term, the Sunset .. I should just look it up once and for all) and also when it was dark but still a bit of light. I'll see if I can find some of the better ones to share with you though not in this issue of course.

**Comment by xexyl** [comment]
This task is proving to be really hard, the moving the entries. It's doing my head in. Not helping is the poor nights. I'm not even sure how to get this all done.

Now the (expectedly) linter is complaining.

One thing I wonder about is if the questions should even be numbered or if they should be an unordered list. I mean what's the point of having the numbers there esp when not in bold and not big text? The `Q: ...` is bigger and stands out more. Might just be better to have the `*` or whatever we have the unordered list character as. I think it would look better, personally.

QUESTION

What do you think about making it unordered instead of ordered (numbered)?

That would be easy to change but t he other part .. that I don't know how to begin. Today I don't think I can do more than I already have I am afraid. Soon I'll probably (or possibly) be afk a bit but then when back I have other things I should take care of.

**Comment by lcn2** [comment]
> This task is proving to be really hard, the moving the entries. It's doing my head in. Not helping is the poor nights. I'm not even sure how to get this all done.
> 
> Now the (expectedly) linter is complaining.
> 
> One thing I wonder about is if the questions should even be numbered or if they should be an unordered list. I mean what's the point of having the numbers there esp when not in bold and not big text? The `Q: ...` is bigger and stands out more. Might just be better to have the `*` or whatever we have the unordered list character as. I think it would look better, personally.
> 
> QUESTION
> 
> What do you think about making it unordered instead of ordered (numbered)?
> 
> That would be easy to change but t he other part .. that I don't know how to begin. Today I don't think I can do more than I already have I am afraid. Soon I'll probably (or possibly) be afk a bit but then when back I have other things I should take care of.

If you are going to use an unordered instead, then we recommend that you revert to using just Markdown unordered lists and FAQ markdown level 2 and 3 headings and not HTML for the table of contents and HTML for the FAQs that follow.

If you do that (reverting the use of HTML lists and instead using just markdown), them we recommend you focus on getting the Table of Contents into the order you wish and then putting the FAQs into the same order.

**Comment by xexyl** [comment]
> > This task is proving to be really hard, the moving the entries. It's doing my head in. Not helping is the poor nights. I'm not even sure how to get this all done.
> > Now the (expectedly) linter is complaining.
> > One thing I wonder about is if the questions should even be numbered or if they should be an unordered list. I mean what's the point of having the numbers there esp when not in bold and not big text? The `Q: ...` is bigger and stands out more. Might just be better to have the `*` or whatever we have the unordered list character as. I think it would look better, personally.
> > QUESTION
> > What do you think about making it unordered instead of ordered (numbered)?
> > That would be easy to change but t he other part .. that I don't know how to begin. Today I don't think I can do more than I already have I am afraid. Soon I'll probably (or possibly) be afk a bit but then when back I have other things I should take care of.
> 
> If you want an unordered list then we recommend that you revert to using just Markdown unordered lists and FAQ markdown headings for the FAQs that follow (as they are now). With an unordered list, we don't need the extra complication of trying that in HTML when markdown can do an unordered list much easier.
> 
> If you do that (reverting the use of HTML lists and instead using just markdown), then we recommend you focus on getting the Table of Contents into the order you wish and then putting the FAQs into the same order.
> 
> ## UPDATE 0a
> If you do the above, and if we have the time, we will look at the final result and if needed make any minor adjustments that might be needed.
> 
> Of course the `<div id="something"> .. </div>` would remain. II is all HTML list stuff that should be reverted in favor of markdown unordered lists: assuming you wish to go with unordered lists.
> 
> We hope this suggestion helps.

Well I'm not sure. The problem with that is the formatting requirements. Unless of course if it's unordered it does not matter as long as the `-` is there. But I think it will actually matter.

It would be nicer if it did not however. Only downside is that it would be a pain to fix. Well maybe. I guess I could delete lines that have `<ol>` and `</ol>` and then delete lines that have `</li>`. Problem is the `<li>` lines have to be manually updated by merging the line below it. But that's assuming it'll even work and I'm not sure if it will.

Anyway I'm afraid I cannot do that now.

The table of contents is one thing. That would be easy to do. But the rest of it I'm not as sure about.


> 
> BTW: We are making progress on the submit server. We also have python developer who is going to spend a bit of time with us this afternoon.

Nice. Good luck! But be safe! Unless of course you're doing it over the Internet of course. I'll be doing other things now I'm afraid. Unsure if I can continue with this issue or any other today. I'm thinking it might have to be an earlier night tonight too; I'm so tired.

Anyway good luck with the submit server! When you have a mailing list sorted I will make as many email accounts as you need for the testing and you can then tell me how to help.

**Comment by lcn2** [comment]
The meeting is via video.

Based on your comments in [GH-issuecomment-2424101903](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2424101903), we suggest you remove the HTML list stuff and just use markdown.

Then get the Table of Contents order in the way you wish.

Then move the FAQs that follow into the same order.

**Comment by xexyl** [comment]
> The meeting is via video.
> 

Okay that's good! Don't want to be too close to a snake in the grass :-)


> Based on your comments in [GH-issuecomment-2424101903](https://github.com/ioccc-src/temp-test-ioccc/issues/2682#issuecomment-2424101903), we suggest you remove the HTML list stuff and just use markdown.
> 

Well that is assuming it works. I'm not sure if it will. It might be that the indentation level is wrong for instance. But that might or might not matter. Not sure. It might end up where each list item is the sole list item per list. That. might not matter with unordered lists but I think it would be a different indentation level without indenting almost all the text. However that probably does not matter and in some ways it might be nice to not have it indented. In some ways it would be nice to have it indented, though. On the other hand it wasn't before so it probably is just fine without being indented. 

Anyway I'll give this a go next time I work on it. Unsure when that will be. If not today I expect tomorrow morning. 

> Then get the Table of Contents order in the way you wish.
> 

I would do that first. That's the easy part. The other part is harder or at least more tedious.

> Then move the FAQs that follow into the same order.

That's the problem, moving the FAQs. It's really doing my head in badly. I'm not sure how to go about that part. But have to get the other things decided first I guess.

A benefit of doing it the markdown way is that the linter would be fine (or so I hope and think).

Anyway best wishes with the meeting!

**Comment by xexyl** [comment]
I tried to look at the markdown in the FAQ but I can't do it right now sadly. I think I'll have to look at it tomorrow. Not sure how that's going to go either. Going to get food soon and after that just relax for the rest of the day. Maybe watch a film or something but undecided.

Hope the meeting went well!

**Comment by lcn2** [comment]
Please withhold PRs related to this issue as we are editing `faq.md` ...

**Comment by xexyl** [comment]
> 



> Please withhold PRs related to this issue as we are editing `faq.md` ...

I'm not doing any. Though I did fix the list problem with another commit - the `<ul>`/`</ul>` that I forgot.

I'm actually going to do other things now but thanks for the help and the update!

**Comment by lcn2** [comment]
With commit d95a55cbfce3f0be6657a6e73092f7c59de448b6 the FAQ entries have been reordered.

Added missing "Why were some filenames changed?" FAQ to the Table of Contents.

Fixed a number of FAQ headers to use the "Q:" at the start.

Added double 1/2 lines between sections.

Made use of double empty lines consistent.

Added some missing "Jump to top" lines.

Shortened 2 very long FAQ long titles.

Perform `make faq` to perform the above and test under macOS.

**NOTE**: There are a few HTML errors detected by the HTML validator, which we are trying to fix now.  **STAY TUNED FOR ANOTHER UPDATE**.

**Comment by xexyl** [comment]
> With commit [d95a55c](https://github.com/ioccc-src/temp-test-ioccc/commit/d95a55cbfce3f0be6657a6e73092f7c59de448b6) the FAQ entries have been reordered.
> 
> Added missing "Why were some filenames changed?" FAQ to the Table of Contents.
> 
> Fixed a number of FAQ headers to use the "Q:" at the start.
> 
> Added double 1/2 lines between sections.
> 
> Made use of double empty lines consistent.
> 
> Added some missing "Jump to top" lines.
> 
> Shortened 2 very long FAQ long titles.
> 
> Perform `make faq` to perform the above and test under macOS.
> 
> **NOTE**: There are a few HTML errors detected by the HTML validator, which we are trying to fix now. **STAY TUNED FOR ANOTHER UPDATE**.
> 
> ## UPDATE 0
> We used two vim sessions, macros, a transfer file, and a 3rd window with the table of contents lines.

Thanks! I'll hold off. It's possible I won't get back to it tomorrow. Really appreciate the assistance here!

We can also talk about answers that should be shortened or revised in other ways, tomorrow.

**Comment by xexyl** [comment]
> 
> ## UPDATE 0
> We used two vim sessions, vim macros, a transfer file, and a 3rd window with the table of contents lines.

Nice. I never really got into vim macros. Not because they're not useful. They certainly are. But because I rarely need them. I often have more than one window open too (`:new` usually) as otherwise if you have the same file open in the same session it will take you to where the other one is.

What do you mean by transfer file, if you get a chance to explain (which is hardly important)?

**Comment by lcn2** [comment]
> > ## UPDATE 0
> > We used two vim sessions, vim macros, a transfer file, and a 3rd window with the table of contents lines.
> 
> Nice. I never really got into vim macros. Not because they're not useful. They certainly are. But because I rarely need them. I often have more than one window open too (`:new` usually) as otherwise if you have the same file open in the same session it will take you to where the other one is.
> 
> What do you mean by transfer file, if you get a chance to explain (which is hardly important)?

Use a vim macro to write a block between two marks to a file in one session that then deletes that block.  Use a vim macro in the other vim session to append the file onto the end of the text.

**Comment by xexyl** [comment]
> > > ## UPDATE 0
> > > We used two vim sessions, vim macros, a transfer file, and a 3rd window with the table of contents lines.
> > 
> > 
> > Nice. I never really got into vim macros. Not because they're not useful. They certainly are. But because I rarely need them. I often have more than one window open too (`:new` usually) as otherwise if you have the same file open in the same session it will take you to where the other one is.
> > What do you mean by transfer file, if you get a chance to explain (which is hardly important)?
> 
> Use a vim macro to write a block between two marks to a file in one session that then deletes that block. Use a vim macro in the other vim session to append the file onto the end of the text.

Ah. I'm afraid that that is beyond me at this moment. Thanks for explaining though! and for doing the moving of text. Normally I can do just fine but there was so much and it's so error prone that it was too much for me at this time.

Anyway I appreciate it a lot and I can continue tomorrow. Have a great rest of your day!

**Comment by lcn2** [comment]
With commit 2b38125ddfe9e996f3f1319f8519aa01c3eb8f61 the HTML validation errors have been fixed for `faq.html` (via `faq.md`).

> Anyway I appreciate it a lot and I can continue tomorrow. Have a great rest of your day!

Since you are leaving for the day, we have another idea we will try to make it easier for people to see where they are while in the middle of the file.  Stay tuned.

**Comment by xexyl** [comment]
> With commit [2b38125](https://github.com/ioccc-src/temp-test-ioccc/commit/2b38125ddfe9e996f3f1319f8519aa01c3eb8f61) the HTML validation errors have been fixed for `faq.html` (via `faq.md`).
> 
> > Anyway I appreciate it a lot and I can continue tomorrow. Have a great rest of your day!
> 
> Since you are leaving for the day, we have another idea we will try to make it easier for people to see where they are while in the middle of the file. Stay tuned.

Sounds great! Well I'll be here for a bit but have other things I need to do too. After you've done your work I can work on it better.

I'll close the pull request I opened earlier on.

**Comment by lcn2** [comment]
With commit aa16ecde4ed0c18b4c6a314cc31a75da6bd1acff we added "Q x.y:" to the Table of Contents and FAQ titles, in part to verify the order, and in part to help someone find their way while inside the FAQ content.

Fixed some of minor inconsistencies in the `faq.md` markdown.

Reduced several FAQ titles that were wrapping onto a 2nd line at the standard window width.

**Comment by xexyl** [comment]
I will work on it more tomorrow. Thanks for all the work!

I will let you know when it seems to be good. Hopefully now that things are ordered right it'll be easy enough to go through.

Hope you have a great rest of your day!

**Comment by xexyl** [comment]
I just did a few minor commits.

I think that FAQs 6.0, 6.1 and 6.2 can be merged somehow. I'll look at that tomorrow. Unfortunately this does mean reordering it which is why I was not in favour of numbering the FAQs. But I guess it won't be that bad. I do have to check the references to the FAQ for these but one thing I can do is use all of the anchor names for the single (or however many) questions.

I believe also that doing this merge the answer can be greatly simplified and shortened.

However I will not look at anything more here until tomorrow.

**Comment by xexyl** [comment]
Expect a MASS file update. All the files with a menu are changed. This is because when the term `web site` was changed to `website` and the `fix_web_site` in the faq.html was changed to `fix_website` the rest of the html files (including those in `inc/`) did not get updated.

I have already merged two FAQs and shortened them too. I am doing more work now. I do not know if I can but I hope to finish this today but if not today then tomorrow.

**Comment by lcn2** [comment]
# REQUEST

Lets plan on closing this issue on 2024 Oct 27 Pacific time some time.

Perhaps it is best if we **soft freeze** the `faq.md` file (i.e., try to not make any substantial content related changes)  on 2024 Oct 25 so that it may have a final check on 2024 Oct 26 along with any PRs that might result from that final check.

**Comment by xexyl** [comment]
> # REQUEST
> Lets plan on closing this issue on 2024 Oct 27 Pacific time some time.
> 
> Perhaps it is best if we **soft freeze** the `faq.md` file (i.e., try to not make any more substantial content related changes ) on 2024 Oct 25 so that it may have a final check on 2024 Oct 26 along with any PRs that might result from that final check.

Just replied to this comment in the other repo. Will do my best to finish the FAQ by then.

**Comment by xexyl** [comment]
I was hoping to work on this issue again today but I am too tired so I'll resume tomorrow. Just as an FYI.

**Comment by lcn2** [comment]
Now is the time to do any last-minute fine-tuning of FAQ.

We plan on adding some text to the FAQ relating to how to register, and the FAQ relating to how to submit an entry.

We plan to put descriptive text, and then put a "place an image" here placeholder for those two FAQs.  This is all part of the process of getting the registration and the submit server in working order.

The editing of those 2 FAQs shouldn't impact what you might do in terms of fine-tuning the FAQ.

**Comment by xexyl** [comment]
# QUESTION

You suggested that the FAQ might be in good enough condition to not worry about it any more. Is this correct or should I do more work here?

I will go take care of other things now but depending on what you have to say here I'll continue with this. Of course when you're gone I can check links and maybe other html files too (though I guess I will do that tomorrow also).

Anyway if you think that this issue is in good shape then I'll not worry about it and instead I'll focus on the other things.

**Comment by lcn2** [comment]
> QUESTION
> You suggested that the FAQ might be in good enough condition to not worry about it any more. Is this correct or should I do more work here?

# ANSWER 

We think it's good enough in general.

We will need to modify 2 entries: 1 for how to register and 1 for how to submit. When that happens, a review of those 2 modified entries would be helpful when that happens.

**Comment by xexyl** [comment]
> > QUESTION
> > You suggested that the FAQ might be in good enough condition to not worry about it any more. Is this correct or should I do more work here?
> 
> # ANSWER 
> 
> We think it's good enough in general.
> 
> We will need to modify 2 entries: 1 for how to register and 1 for how to submit. When that happens, a review of those 2 modified entries would be helpful when that happens. 

Sounds good. 

So then tomorrow will be wrt the IOCCC pretty easy. I don't really know what other files to check except the bin/index.html file so I will start with that and then see how it goes. 

Thanks.

**Comment by lcn2** [comment]
We plan to improve the FAQ 0.0 sub-section (2) "_Register for the IOCCC_" and (5) "_Upload your entry to the IOCCC submit server_" to replace the "_is still being written"_ stuff with a little more information.

Once that is done, we plan to close this issue.

But first, we need to get ready to work with our python helped on the submit-server tomorrow.

**Comment by xexyl** [comment]
> We plan to improve the FAQ 0.0 sub-section (2) "_Register for the IOCCC_" and (5) "_Upload your entry to the IOCCC submit server_" to replace the "_is still being written"_ stuff with a little more information.
> 
> 
> 
> Once that is done, we plan to close this issue.
> 
> 

Want me to look at those two as well?

**Comment by xexyl** [comment]
> 
> But first, we need to get ready to work with our python helped on the submit-server tomorrow.

Good luck with that! Always dangerous playing with snakes 🐍!Even when it's not venomous they can still easily kill you .. or in the case of Python: fry your brain 🧠!

**Comment by lcn2** [comment]
# NOTICE

See [GH-issuecomment-2439833011](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2439833011) as that comment applies to this issue as well.

**Comment by xexyl** [comment]
> # NOTICE
> See [GH-issuecomment-2439833011](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2439833011) as that comment applies to this issue as well.

Thanks for the 🤯💀🆙

I replied over there. I think I'm all good until a final spot check! I made a small pull request but I can't think of anything else now, begging your pardon (as the expression goes :-) ).

**Comment by lcn2** [comment]
We will now close this issue with our **THAKS 🙏**, @xexyl.  People will benefit from the improved FAQ.  Thank you!

p.s. As noted in [GH-ssuecomment-2439833011](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2439833011), as part of the TODO items of issue #2239 :

> updates to FAQ that relate to the registration process or the submission process, etc.) will be "happily entertained" as the expression goes

I.e., as part of performing the **Great Fork Merge** TODO items, we will modify FAQ 0.0 subsection 2 "_Register for the IOCCC_" and subsection 5 "_Upload your entry to the IOCCC submit server_", in addition to the other edits as noted in the TODOs of issue #2239.
